Test/CoreSDK.Test/Shared/TelemetryClientTest.cs

```diff
@@ -494,29 +498,42 @@ public void ChannelIsInitializedInTrackWhenTelemetryConfigurationIsConstructedVi
         [TestMethod]
         public void TrackUsesInstrumentationKeyIfSetInCodeFirst()
         {
+            ClearActiveTelemetryConfiguration();
             PlatformSingleton.Current = new StubPlatform();
+            string message = "Test Message";
 
             ITelemetry sentTelemetry = null;
             var channel = new StubTelemetryChannel { OnSend = telemetry => sentTelemetry = telemetry };
             var configuration = new TelemetryConfiguration { TelemetryChannel = channel };
             var client = new TelemetryClient(configuration);
 
+            string oldActiveKey = TelemetryConfiguration.Active.InstrumentationKey;
+            string environmentKey = Guid.NewGuid().ToString();
             string expectedKey = Guid.NewGuid().ToString();
-            client.Context.InstrumentationKey = expectedKey; // Set in code
-            configuration.InstrumentationKey = Guid.NewGuid().ToString(); // Set in config
-            Environment.SetEnvironmentVariable("APPINSIGHTS_INSTRUMENTATIONKEY", expectedKey); // Set via env. variable
-            Assert.DoesNotThrow(() => client.TrackTrace("Test Message"));
+            Environment.SetEnvironmentVariable("APPINSIGHTS_INSTRUMENTATIONKEY", environmentKey); // Set via the environment variable.
 
+            // Set in code method one.
+            client.Context.InstrumentationKey = expectedKey;
+            Assert.DoesNotThrow(() => client.TrackTrace(message));
+            Assert.Equal(expectedKey, sentTelemetry.Context.InstrumentationKey);
+
+            // Set in code method two.
```

Comment:
I did that originally, it was just a lot of copied and pasted code, and the test naming got weirder but I can do it again.